### DISCUSSION TITLE
read/write to device storage on Android 11 - help or advice needed

### DISCUSSION THREAD

**Discussion by trinoo1979** [airsdk/Adobe-Runtime-Support#1548] [General]
Hello guys, first of all, Im sorry if this is an off topic here - not sure where to post it as this forum is the only AIR forum I know about...

Problem (in more general): 
I have an apllication on Google Play, which stores data in a small SQLite database. My application includes a feature that allows the user to export the database (for backing-up purposes and also to easily migrate to a different device) which means application copies its DB file from internal app storage not accessible by the user to location, that user can access = root of internal device storage. App also in opposite allows to import DB file, which means it copies DB file (placed somehere in the device by the user, downloaded by him or transferred from different device) from user accessible storage back to internall app storage. To easily pick-up DB file when importing it my app includes a small built-in file browser.
 
Until Android 11 all worked fine. On Android 11 my import export functions do not work. The root of internal device storage is not accessible on Android 11 if I understand right (which can be probably solved finding the different location or folder to write a copy of my database file, something similar used here in [#1343 ](https://github.com/airsdk/Adobe-Runtime-Support/discussions/1343) but more crucial is, that I have no idea, how to get also my import function back to work. On Android 11 my built-in file browser shows up, but I can only see media files in it, not a .db file.

I was trying to read all Google documents related to the new Android 11 policies and restrictions a few times (scoped storage access, permissions, etc.), but as an AS3/AIR (not Android) programmer I simple dont understand, how to handle them. 

So please, is it my approach (import/export) still applicable on Android 11? Can I access a non media file placed in Document or Downloaded folder from my app? Or is it possible to invoke Android internal file picker for that task (to pick a file)?
Can any of you, please, bring the light and give me any suggestions or point me to any simple AS3 AIR for Android tutorial, how to handle read/write/share file to and from user accessible storage? Or is there any ANE I can use for that purpose?

**Comment by teotigraphix** [comment]
See this;

https://forum.starling-framework.org/d/21307-android-q-file-system/38
